src/pudl/transform/ferc1.py

```diff
@@ -1188,11 +1196,15 @@ class PlantInServiceFerc1TableTransformer(Ferc1AbstractTableTransformer):
     @cache_df(key="dbf")
     def align_row_numbers_dbf(self, df: pd.DataFrame) -> pd.DataFrame:
         """Align historical FERC1 DBF row numbers with XBRL account IDs."""
-        return pd.merge(
-            df,
-            get_ferc1_dbf_xbrl_glue(self.source_table_id(Ferc1Source.DBF)),
-            on=["report_year", "row_number"],
-        ).rename(columns={"xbrl_column_stem": "ferc_account_label"})
+        dbf_to_xbrl_map = fill_dbf_to_xbrl_map(
+            df=read_dbf_to_xbrl_map(
+                dbf_table_name=self.source_table_id(Ferc1Source.DBF)
+            ),
+            dbf_years=Ferc1Settings().dbf_years,
+        )
+        return pd.merge(df, dbf_to_xbrl_map, on=["report_year", "row_number"]).rename(
+            columns={"xbrl_column_stem": "ferc_account_label"}
+        )
```

Comment:
should this merge be a left merge? just thinking about the potential to silently loose records from `df`.

Reply:
I switched this to a left merge and got a bunch of new "duplicate" records. But they are duplicates because they're entirely NULL records. They're the records that correspond to the header rows, which we actually *do* want to lose here, since they should never contain data, and generally do not contain data.

The DBF-XBRL mapping is effectively **defining** what DBF row numbers have a known meaning, so I think we do want an inner merge here. However some kind of check to make sure that there's no data in any rows that exist outside of this definition of known meaning would also be good... what would that look like?

We also know that there are a few cases in which a utility reports year N data using the form from year N-1, and that results in data showing up in the wrong rows (including header rows). Ideally we would be able to recover that data by parsing it using the appropriate form-year mapping (and being able to identify what form-year they are using from the pattern of NA filled header rows). But that's beyond the scope of the current work.

What do you think  we should do here?

Reply:
* check if `dbf_to_xbrl_map` has no nulls in `xbrl_column_stem`
* do left merge
* check if merged `df` has any na's in `ferc_account_label` (renamed from `xbrl_column_stem`